javascript/modules/client/test/integration/client-addressList.test.ts

```diff
@@ -0,0 +1,405 @@
+            );
+        }
+      }
+
+    })
+  })
+
+  describe('Action generators', () => {
+    it("Should create a Mulisig client and generate a init action", async () => {
+      const withdrawParams: IAddressListPluginInstall = {
+        proposals: {
+          minDuration: 7200, // seconds
+          minTurnout: 0.5,
+          minSupport: 0.5
+        },
+        addresses: [
+          "0x1234567890123456789012345678901234567890",
+          "0x2345678901234567890123456789012345678901",
+          "0x3456789012345678901234567890123456789012",
+          "0x4567890123456789012345678901234567890134"
+        ]
+      };
+
+      const initAction = ClientAddressList.encoding.installEntry(withdrawParams);
+
+      expect(typeof initAction).toBe("object");
+      // what does this should be
+      expect(initAction.data).toBeInstanceOf(Uint8Array);
+    });
+
+    it("Should create a Mulisig client and generate a plugin config action action", async () => {
+      const context = new ContextPlugin(contextParamsLocalChain);
+      const client = new ClientAddressList(context);
+
+      const pluginConfigParams: ProposalConfig = {
+        minDuration: 100000,
+        minTurnout: 0.25,
+        minSupport: 0.51
+      };
+
+      const initAction = client.encoding.setPluginConfigAction(pluginConfigParams);
+
+      expect(typeof initAction).toBe("object");
+      // what does this should be
+      expect(initAction.data).toBeInstanceOf(Uint8Array);
+    });
+  })
+
+  describe('Data retrieval', () => {
+    it("Should get the list of members that can vote in a proposal", async () => {
+      const context = new ContextPlugin(contextParamsLocalChain);
+      const client = new ClientAddressList(context);
+
+      const wallets = await client.methods.getMembers()
+
+      expect(Array.isArray(wallets)).toBe(true);
+      expect(wallets.length).toBeGreaterThan(0);
+      expect(typeof wallets[0]).toBe('string');
+      expect(wallets[0]).toMatch(/^0x[A-Fa-f0-9]{40}$/i);
+    })
+    it("Should fetch the given proposal", async () => {
+      const context = new ContextPlugin(contextParamsLocalChain);
+      const client = new ClientAddressList(context);
+
+      const proposalId = "0x1234567890123456789012345678901234567890_0x55"
+      const proposal = await client.methods.getProposal(proposalId)
+
+      expect(typeof proposal).toBe('object');
+      expect(proposal.id).toBe(proposalId);
+      expect(proposal.id).toMatch(/^0x[A-Fa-f0-9]{40}_0x[A-Fa-f0-9]{1,}$/i);
+    })
+    it("Should get a list of proposals filtered by the given criteria", async () => {
+      const context = new ContextPlugin(contextParamsLocalChain);
+      const client = new ClientAddressList(context);
+      const limit = 2
+      const params: IProposalQueryParams = {
+        limit
+      }
+      const proposals = await client.methods.getProposals(params)
+
+      expect(Array.isArray(proposals)).toBe(true)
+      expect(proposals.length <= limit).toBe(true)
```

Comment:
I would test different limits
- 0 => using the default value (getting 10 results)
- 1 => getting 1 entry
- 5 => getting 5 entries

The UI may be using this now